pkg/file/file.go

```diff
@@ -11,6 +11,22 @@ import (
 	"github.com/fleetdm/fleet/v4/pkg/secure"
 )
 
+// ExtractInstallerMetadata extracts the software name and version from the
+// installer file. The format of the installer is determined based on the
+// extension of the filename.
+func ExtractInstallerMetadata(filename string, b []byte) (name, version string, err error) {
+	switch ext := filepath.Ext(filename); ext {
+	case ".deb":
+		return ExtractDebMetadata(b)
+	case ".exe":
+		return ExtractPEMetadata(b)
+	case ".pkg":
+		return ExtractXARMetadata(b)
+	default:
+		return "", "", fmt.Errorf("unsupported file type: %s", ext)
+	}
```

Comment:
.msi support will come in a subsequent PR. Also, I don't think it makes sense for me to address the "try to match it to a db record" part with this ticket, it will be better served if done when / where it is needed (e.g. maybe we'll want a "MatchOrCreate" datastore method, will be clearer what to receive as args, etc.). I'd cut this part from the ticket.